packages/shared/src/hooks/useOrganizationList.tsx

```diff
@@ -1,39 +1,191 @@
 import type {
+  ClerkPaginationResponse,
   CreateOrganizationParams,
+  GetUserOrganizationInvitations,
   OrganizationMembershipResource,
   OrganizationResource,
   SetActive,
+  UserOrganizationInvitationResource,
+  UserResource,
 } from '@clerk/types';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import useSWRInfinite from 'swr/infinite';
 
 import { useClerkInstanceContext, useUserContext } from './contexts';
 
+type UseOrganizationListParams = {
+  userInvitations?:
+    | true
+    | (GetUserOrganizationInvitations & {
+        aggregate: boolean;
+      });
+};
+
 type OrganizationList = ReturnType<typeof createOrganizationList>;
 
 type UseOrganizationListReturn =
-  | { isLoaded: false; organizationList: undefined; createOrganization: undefined; setActive: undefined }
   | {
-      isLoaded: true;
+      isLoaded: false;
+      organizationList: undefined;
+      createOrganization: undefined;
+      setActive: undefined;
+      userInvitations: {
+        data: undefined;
+        count: number;
+        isLoading: false;
+        isValidating: false;
+        isError: false;
+        size: undefined;
+        setSize: undefined;
+      };
+    }
+  | {
+      isLoaded: boolean;
       organizationList: OrganizationList;
       createOrganization: (params: CreateOrganizationParams) => Promise<OrganizationResource>;
       setActive: SetActive;
+      userInvitations: {
+        data: UserOrganizationInvitationResource[];
+        count: number;
+        isLoading: boolean;
+        isValidating: boolean;
+        isError: boolean;
+        size: number | undefined;
+        setSize:
+          | ((
+              size: number | ((_size: number) => number),
+            ) => Promise<(ClerkPaginationResponse<UserOrganizationInvitationResource> | undefined)[] | undefined>)
+          | undefined;
+      };
     };
 
-type UseOrganizationList = () => UseOrganizationListReturn;
+type UseOrganizationList = (params?: UseOrganizationListParams) => UseOrganizationListReturn;
 
-export const useOrganizationList: UseOrganizationList = () => {
+export const useOrganizationList: UseOrganizationList = params => {
+  const { userInvitations } = params || {};
+
+  const triggerInfinite = userInvitations !== true ? userInvitations?.aggregate ?? false : false;
```

Comment:
Due to the type of `userInvitations`, this check is a little confusing. Let's make it more explicit:

```suggestion
  const triggerInfinite = typeof userInvitations === 'boolean' && userInvitations ? false : 
!!userInvitations?.aggregate;
```

(please double-check the above for correctness)